packages/payment-backend/src/service.ts

```diff
@@ -219,6 +219,24 @@ export const getOrderPaymentFilters = (p: { orderId: string }) => {
   })
 }
 
+export const filterPaymentMethodByGateway = (
+  filteredPaymentFilters: any[],
+  globallyFilteredPaymentGateways: string[]
+) =>
+  filteredPaymentFilters.filter((method) => {
```

Comment:
you need to add "return" so that the function returns the filtered array. filter -function does not do filtering in-place.

Reply:
Never mind, I did not notice it was lambda function, this is ok!